concordium-consensus/src/Concordium/Scheduler/ProtocolLevelTokens/Module.hs

```diff
@@ -90,254 +94,249 @@ initializeToken tokenParam = do
+                                              trrReason = Just "recipient in deny list"
+                                            }
+                            pltTickEnergy tokenTransferCost
+                            success <- transfer tcSender recipientAccount pthoAmount pthoMemo
+                            unless success $ do
                                 availableBalance <- getAccountBalance tcSender
                                 failTH
                                     TokenBalanceInsufficient
                                         { trrOperationIndex = opIndex,
                                           trrAvailableBalance = toTokenAmount decimals availableBalance,
-                                          trrRequiredBalance = ptgoUnprocessedAmount
+                                          trrRequiredBalance = ttAmount pthoUnprocessed
                                         }
-                        PTGOTokenAddAllowList{..} -> do
-                            requireFeature opIndex "addAllowList" "allowList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "allowList" (Just "")
-                            logEncodeTokenEvent (AddAllowListEvent ptgoTarget)
-                        PTGOTokenRemoveAllowList{..} -> do
-                            requireFeature opIndex "removeAllowList" "allowList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "allowList" Nothing
-                            logEncodeTokenEvent (RemoveAllowListEvent ptgoTarget)
-                        PTGOTokenAddDenyList{..} -> do
-                            requireFeature opIndex "addDenyList" "denyList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "denyList" (Just "")
-                            logEncodeTokenEvent (AddDenyListEvent ptgoTarget)
-                        PTGOTokenRemoveDenyList{..} -> do
-                            requireFeature opIndex "removeDenyList" "denyList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "denyList" Nothing
-                            logEncodeTokenEvent (RemoveDenyListEvent ptgoTarget)
+                        tokenGovernanceOp -> do
+                            mbGovAccountIx <- getTokenState "governanceAccount"
+                            case mbGovAccountIx of
+                                Nothing -> error "Invariant violation: Token state is missing the governance account address."
```

Comment:
I _really_ want to avoid the possibility of `error`s in the token module. If the governance account is missing, then I think this should be denied as `OperationNotPermitted` since if there is no governance account then trivially the sender is not the governance account.